lib/Pipedrive.js

```diff
@@ -28,6 +30,71 @@
 			callback(error, collectionItems, additionalData);
 		}, false);
 	};
+	exports.OAuthAuthenticate = function (client_id, redirect_uri, res) {
+	    res.redirect('https://oauth.pipedrive.com/oauth/authorize?' +
+            querystring.stringify({
+                client_id: client_id,
+                redirect_uri: redirect_uri,
+            }))
+	};
+
+	exports.OAuthAuthenticateCallback = function (req, res, redirect_uri, clientId, clientSecret) {
+	    var code = req.query.code || null;
+	    var authOptions = {
+	        url: 'https://oauth.pipedrive.com/oauth/token',
+            form: {
+	            code: code,
+                redirect_uri: redirect_uri,
+                grant_type: 'authorization_code'
+            },
+            headers: {
+	            'Authorization': 'Basic ' + (new Buffer(
+	                clientId + ':' + clientSecret
+                ).toString('base64'))
+            },
+            json: true
+	    };
+	    return new Promise(function (resolve, reject) {
+	        request.post(authOptions, function (error, response, body) {
+	            if (error) {
+	                log('OAuthAuthenticateCallback.error handling: ' + error);
+	                reject(error)
```

Comment:
Just small note, that `reject` doesn't finish function execution. For example:

```javascript
new Promise(function (resolve, reject) {
    reject(new Error("BOOM!"));
    console.log("hello world");
});
```

It will prints `hello world`. So it's reliable to make `return` after `reject(error)` or `return reject(error)` in order to make code execution clear.

Reply:
and seems, missed `;`, eslint can be very helpful ;)